ivadomed/transforms.py

```diff
@@ -278,6 +281,102 @@ def __call__(self, sample, metadata=None):
         return data_out, metadata
 
 
+class Resize(ImedTransform):
+    """
+    Resize image to a given resolution.
+
+    Args:
+        height (int): Resolution along the first axis, in pixel.
+        width  (int): Resolution along the second axis, in pixel.
+        interpolation_order (int): different strategies for interpolation. 
+    """
+
+    def __init__(self, height, width, interpolation=None):
+        self.height = height
+        self.width  = width
+
+    @multichannel_capable
+    @two_dim_compatible
+    def undo_transform(self, sample, metadata=None):
+        """Resize to original resolution."""
+        assert "data_shape" in metadata
+        
+        # Get params
+        original_shape = metadata["preresize_shape"]
+        current_shape = sample.shape
+        
+        # Undo resampling
+        data_out = resize(sample, original_shape[:2])
+
+        # Data type
+        data_out = data_out.astype(sample.dtype)
+
+        return data_out, metadata
+
+    @multichannel_capable
+    @multichannel_capable  # for multiple raters during training/preprocessing
+    @two_dim_compatible
+    def __call__(self, sample, metadata=None):
+        """Resize to a given resolution, in pixel."""
+        # Get params
+        # Voxel dimension in mm
+        is_2d = sample.shape[-1] == 1
+        metadata['preresize_shape'] = sample.shape
+        
+        # Run resizing
+        data_out = resize(sample, (self.height, self.width))
+
+        # Data type
+        data_out = data_out.astype(sample.dtype)
+        return data_out, metadata
+
+
+class VertebralSplitting(ImedTransform):
```

Comment:
is **transforms.py** the right location for this function?

Reply:
VertebralSplitting transforms the mask into separate joints, thus, it is a transformation function.

Reply:
right, but this is not specific to vertebral pose estimation-- e.g. it could be applied to any rotational node, right? in that case i suggest to generalize the naming conventions for better clarity and reusability

Reply:
It is not specific for vertebral, it can be applied on any segmentation mask which contains several dis-connected components and we want to separate them. I did not change the name but it can be changed.